lib/data/array.ml

```diff
@@ -111,14 +118,43 @@ instance eq 'a => eq (array 'a) begin
       loop a.offset
 end
 
-let ( .() ) (Array r as arr) i =
-  in_bounds "(.())" arr i
-  geti r.backing (i + r.offset)
-
-let ( .()<- ) (Array r as arr) i x =
-  in_bounds "update" arr i
-  let _ = seti r.backing (i + r.offset) x
-  ()
+let empty = Array { length = 0, offset = 0, backing = empty_storage }
```

Comment:
I really don't feel safe sharing a single table as backing storage for a module of mutable arrays.

Reply:
I get that, but it is perfectly safe - you can't mutate it as it's empty.

Reply:
Aye, if you say so!